src/fmu/dataio/providers/objectdata/_faultroom.py

```diff
@@ -66,11 +68,24 @@ def get_bbox(self) -> BoundingBox3D:
     def get_spec(self) -> FaultRoomSurfaceSpecification:
         """Derive data.spec for FaultRoomSurface"""
         logger.info("Get spec for FaultRoomSurface")
+
+        juxtaposition_ordered_hw = []
+        juxtaposition_ordered_fw = []
+        if isinstance(self.dataio.config, GlobalConfiguration) and (
+            strat := self.dataio.config.stratigraphy
+        ):
+            juxtaposition_ordered_hw = strat.get_ordered_official_names(
+                self.obj.juxtaposition_hw
+            )
+            juxtaposition_ordered_fw = strat.get_ordered_official_names(
+                self.obj.juxtaposition_fw
+            )
```

Comment:
```suggestion
        if isinstance(self.dataio.config, GlobalConfiguration) and (
            strat := self.dataio.config.stratigraphy
        ):
            juxtaposition_hw = [strat.get_official_name(juxt) or juxt for juxt in self.obj.juxtaposition_hw]
            juxtaposition_fw = [strat.get_official_name(juxt) or juxt for juxt in self.obj.juxtaposition_fw]
```

Does this actually need to be ordered, or could we simplify and do this instead 🙂 ? We are planning to provide an ordered stratigraphy as FMU masterdata that will be uploaded to Sumo. I guess consumers could order this themselves? My main concern is about zones/horizons that does not have a smda mapping, what should their position in this list be? 🤷‍♀️

Reply:
```suggestion
        juxtaposition_ordered_hw = []
        juxtaposition_ordered_fw = []
        if isinstance(self.dataio.config, GlobalConfiguration) and (
            strat := self.dataio.config.stratigraphy
        ):
            juxtaposition_ordered_hw =  [strat[item].name for item in strat if item in self.obj.juxtaposition_hw]
            juxtaposition_ordered_fw = [strat[item].name for item in strat if item in self.obj.juxtaposition_fw]

```

Reply:
the suggestion above will give a list ordered by the order in the `stratigraphy`. However as mentioned we need to check that all items in the `juxtaposition_fw` / `juxtaposition_hw` are defined in `stratigraphy` to use this approach 🙂